Source/WebCore/rendering/RenderText.cpp

```diff
@@ -1770,7 +1770,8 @@ float RenderText::width(unsigned from, unsigned length, const FontCascade& fontC
                 // The rare case of when we switch between IFC and legacy preferred width computation.
                 if (!m_maxWidth)
                     width = maxLogicalWidth();
-                width = *m_maxWidth;
+                else
```

Comment:
@alanbaradlay FYI, this is one place where I fixed the code rather than dropping the dead code.

Reply:
Looks like this code would crash or abort with a null dereference before? Any ideas how to make a test case?

Reply:
@alanbaradlay What do you think?

Reply:
not great code (especially the const_cast part) but calling maxLogicalWidth() does assign a value to m_maxWidth.
```
float RenderText::maxLogicalWidth() const
{
    if (preferredLogicalWidthsDirty() || !m_maxWidth)
        const_cast<RenderText*>(this)->computePreferredLogicalWidths(0, !preferredLogicalWidthsDirty());

    return *m_maxWidth;
}

```

Reply:
Oh wow, Ok. So, maybe a a static analysis false positive.

I guess we should refactor this code to make this clearer and hopefully get rid of the static analysis warning.